src/ImageSharp/Processing/IImageProcessingContext.cs

```diff
@@ -15,6 +16,12 @@ public interface IImageProcessingContext
         /// </summary>
         Configuration Configuration { get; }
 
+        /// <summary>
+        /// Gets a set of properties for the Image Processing Context.
+        /// </summary>
+        /// <remarks>This can be used for storing global settings and defaults to be accessable to processors.</remarks>
+        IDictionary<object, object> Properties { get; }
```

Comment:
I wonder if we want to consider something more strongly-typed here?
- `System.Type` or a custom object as a key
- Restrictions for the value: do we expect simple values here? If not, maybe a marker interface.
- Our own types `GraphicsProperties` / `IGraphicsProperties` instead of `IDictionary<K,V>`

Reply:
no it should be explicitly be as loosely typed as possible and we should not be limiting anything from being but in the property bag, that's why I even when as far as using `<object,object>` rather than `<string,object>` to allow developers to use it for there needs they can do anything with it... use strings, use private object instances (no way for someone to accidently reuse the key) or anything in between... we would be loosing one of the key flexibility benefits of this extension point is we make the type a custom type with restrictions..